lib/Analysis/ControlDependenceAnalysis.cpp

```diff
@@ -0,0 +1,333 @@
+//===- ControlDependenceAnalysis.h - Control dependence analyis utilities
+//----------*- C++ -*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Declares some utility functions useful to analyzing the control dependencies
+// between basic blocks of the CFG. The control dependendies are calculated
+// using the algorithm from the following paper
+//   J.Ferrante, K.J. Ottenstein, and J. D. Warren, "The Program Dependence
+//   Graph and its Use in Optimizations", ACM Trans. Program. Lang. Syst., vol.
+//   9, pp. 319-349, 1987.
+//
+//===----------------------------------------------------------------------===//
+
+#include "dynamatic/Analysis/ControlDependenceAnalysis.h"
+#include "llvm/Support/Debug.h"
+
+using namespace mlir;
+using namespace llvm;
+using namespace dynamatic;
+
+#define DEBUG_TYPE "cdg-analysis"
+
+void ControlDependenceAnalysis::identifyAllControlDeps(
+    mlir::func::FuncOp &funcOp) {
+  mlir::Region &funcReg = funcOp.getRegion();
+
+  // Initialize the all_control_deps_map by creating an entry for every block
+  // constituting the region
+  for (mlir::Block &block : funcReg.getBlocks()) {
+    llvm::SmallVector<mlir::Block *, 4> deps;
+    all_control_deps_map.insert(std::make_pair(&block, deps));
+  }
+
+  if (funcReg.hasOneBlock()) {
+    return; // functions made up of 1 basic block do not have any dependencies
+  }
```

Comment:
Here and everywhere: remove the bracket if the if/else/loop block contains only a single statement:

https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements